yt_dlp/extractor/nintendo.py

```diff
@@ -1,57 +1,101 @@
     }]
 
+    def _create_asset_url(self, path):
+        return urljoin('https://assets.nintendo.com/', urllib.parse.quote(path))
+
     def _real_extract(self, url):
-        page_id = self._match_id(url)
+        locale, slug = self._match_valid_url(url).group('locale', 'slug')
+
+        language, _, country = (locale or 'en').partition('-')
+        parsed_locale = f'{language.lower()}_{country.upper() or "US"}'
+        self.write_debug(f'Using locale {parsed_locale} (from {locale})', only_once=True)
+
+        response = self._download_json('https://graph.nintendo.com/', slug, query={
+            'operationName': 'NintendoDirect',
+            'variables': json.dumps({
+                'locale': parsed_locale,
+                'slug': slug,
+            }, separators=(',', ':')),
+            'extensions': json.dumps({
+                'persistedQuery': {
+                    'version': 1,
+                    'sha256Hash': '969b16fe9f08b686fa37bc44d1fd913b6188e65794bb5e341c54fa683a8004cb'
+                },
+            }, separators=(',', ':')),
+        })
+        # API returns `{"data": {"direct": null}}` if no matching id
+        direct_info = traverse_obj(response, ('data', 'direct', {dict}))
+        if not direct_info:
+            raise ExtractorError(f'No Nintendo Direct with id {slug} exists', expected=True)
 
-        webpage = self._download_webpage(url, page_id)
+        errors = ', '.join(traverse_obj(response, ('errors', ..., 'message')))
+        if errors:
+            raise ExtractorError(f'GraphQL API error: {errors or "Unknown error"}')
 
-        entries = [
-            OoyalaIE._build_url_result(m.group('code'))
-            for m in re.finditer(
-                r'data-(?:video-id|directVideoId)=(["\'])(?P<code>(?:(?!\1).)+)\1', webpage)]
+        asset_id = traverse_obj(direct_info, ('video', 'publicId', {str}))
+        if not asset_id:
+            self.raise_no_formats('Could not find any video formats', video_id=slug)
 
-        title = self._html_search_regex(
-            r'(?s)<(?:span|div)[^>]+class="(?:title|wrapper)"[^>]*>.*?<h1>(.+?)</h1>',
-            webpage, 'title', fatal=False)
+        result = {
+            'display_id': slug,
+            'formats': self._extract_m3u8_formats(
+                self._create_asset_url(f'/video/upload/sp_full_hd/v1/{asset_id}.m3u8'), slug),
+        }
+        result.update(traverse_obj(direct_info, {
+            'id': ('id', {str}),
+            'title': ('name', {str}),
+            'timestamp': ('startDate', {unified_timestamp}),
+            'description': ('description', 'text', {str}),
+            'age_limit': ('contentRating', 'order', {int}),
+            'tags': ('contentDescriptors', ..., 'label', {str}),
+            'thumbnail': ('thumbnail', {self._create_asset_url}),
+        }))
+        if asset_id.startswith('Legacy Videos/'):
+            result['_old_archive_ids'] = [make_archive_id(self, remove_start(asset_id, 'Legacy Videos/'))]
```

Comment:
since we're already checking `if str.startswith` we don't need `remove_start`
```suggestion
        if asset_id.startswith('Legacy Videos/'):
            result['_old_archive_ids'] = [make_archive_id(self, asset_id[14:])]
```